### DISCUSSION TITLE
How to pass `preserve_annotations` to terser

### DISCUSSION THREAD

**Discussion by jjenzz** [parcel-bundler/parcel#6055] [Q&A] [answered]
I'm trying to keep the`/*#__PURE__*/` annotations that Parcel adds to React APIs in our minified output. It looks like terser has an option specifically for this but I cannot figure out how to pass this to Parcel. I've tried a `.terserrc` without luck.

```js
{
  "output": {
    "preserve_annotations": true
  }
}
```

Any idea how to achieve this?

**Comment by mischnic** [comment]
I can't get terser itself to preserve them on https://try.terser.org/

config:
```js
{
  module: true,
  compress: false,
  format: {
    preserve_annotations: true
  }
}
```

(the possible options are described here: https://github.com/terser/terser/blob/d2927ef4c8d23846cce2f0d0ae4ec8f17fbe4e5f/tools/terser.d.ts#L138)

input code:
```js
export const X = /*#__PURE__*/ (() => 1);
```

output:
```js
export const X=()=>1;
```

**Reply by benoitgrelard** [reply]
It seems you need `comments: true,` as well

**Reply by mischnic** [reply]
```js
{
  module: true,
  compress: false,
  format: {
    preserve_annotations: true,
    comments: true
  }
}
```
does preserve that one comment, but it also doesn't remove any other comments. I think `comments: "some"` should be changed to not only include `@license` comments but also these PURE comments:
https://github.com/terser/terser#format-options

**Reply by mischnic** [reply]
-> https://github.com/terser/terser/issues/968

**Comment by jjenzz** [comment, marked as answer]
Ooooh look, input from a few more eyes and I've landed here:

```js
{
  module: true,
  compress: false,
  format: {
    preserve_annotations: true,
    comments: '/__PURE__/',
  }
}
```

This works! 🎉  Thanks!
